src/errorpage.cc

```diff
@@ -1348,15 +1348,12 @@ ErrorState::BuildHttpReply()
     // Make sure error codes get back to the client side for logging and
     // error tracking.
     if (request) {
-        if (detail)
-            request->detailError(type, detail);
-        else
-            request->detailError(type, SysErrorDetail::NewIfAny(xerrno));
+        request->error.update(type, detail);
+        request->error.update(SysErrorDetail::NewIfAny(xerrno));
```

Comment:
It is possible to add more or more complex Error methods to handle this better, but I did not do that because this code should not exist at all. There is already a TODO that would refactor how ErrorState handles these error details. When we are done, I suspect that ErrorState will start using Error objects in ALE and/or HttpRequest to handle them, eliminating this error-prone code.